### ISSUE TITLE
Investigate using LTTB for maxDataPoints

### ISSUE THREAD

**Issue by dgryski** [go-graphite/carbonapi#154]
Maybe `consolidateBy=lttb` ?

https://github.com/dgryski/go-lttb

**Comment by Civil** [comment]
Can you provide more details about that?

**Comment by JaderDias** [comment]
@Civil I didn't know what it was either, but I followed the links and got a nice page https://godoc.org/github.com/dgryski/go-lttb

**Comment by Civil** [comment]
Yeah, but I'd like to get more information about what we were expecting from that :)

**Comment by arodland** [comment]
The idea is that it provides a more visually representative view of the data than other downsampling algorithms.

The original paper is at http://skemman.is/stream/get/1946/15343/37285/3/SS_MSthesis.pdf . It seems to have been pretty well adopted in the time since, and you can find some convincing visual results if you search the web.

**Comment by dgryski** [comment]
Two things I ran into when looking into this: 1) figuring out the API and where exactly in carbonapi this would get implemented and 2) LTTB doesn't return evently spaced (in time) points, which means it doesn't follow the start/stop/step pattern (which makes it tricky to fit into the responses we have).  If grafana is happy with this, then we're good.
